lib/include/jxl/decode.h

```diff
@@ -875,6 +853,69 @@ JXL_EXPORT JxlDecoderStatus JxlDecoderSetPreferredColorProfile(
 JXL_EXPORT JxlDecoderStatus JxlDecoderSetDesiredIntensityTarget(
     JxlDecoder* dec, float desired_intensity_target);
 
+/**
+ * Sets the desired output color profile of the decoded image either from a
+ * color encoding or an ICC profile. Valid calls of this function have either @c
+ * color_encoding or @c icc_data set to NULL and @c icc_size must be 0 if and
+ * only if @c icc_data is NULL.
+ *
+ * Depending on whether a color management system (CMS) has been set the
+ * behavior is as follows:
+ *
+ * If a color management system (CMS) has been set with @ref JxlDecoderSetCms,
+ * and the CMS suppports output to the desired color encoding or ICC profile,
+ * then it will provide the output in that color encoding or ICC profile. If the
+ * desired color encoding or the ICC is not supported, then an error will be
+ * returned.
+ *
+ * If no CMS has been set with @ref JxlDecoderSetCms, there are two cases:
+ *
+ * (1) Calling this function with a color encoding will convert XYB images to
+ * the desired color encoding. In this case, if the requested color encoding has
+ * a narrower gamut, or the white points differ, then the resulting image can
+ * have significant color distortion. Non-XYB images will not be converted to
+ * the desired color space.
+ *
+ * (2) Calling this function with an ICC profile will result in an error.
+ *
+ * If called with an ICC profile (after a call to @ref JxlDecoderSetCms), the
+ * ICC profile has to be a valid RGB or grayscale color profile.
+ *
+ * Can only be set after the @ref JXL_DEC_COLOR_ENCODING event occurred and
+ * before any other event occurred, and should be used before getting
+ * JXL_COLOR_PROFILE_TARGET_DATA. The color_encoding or ICC profile must be
+ * grayscale if num_color_channels from the basic info is 1, RGB if
+ * num_color_channels from the basic info is 3.
```

Comment:
What if the image is tagged as grayscale in the Basic Info, but the `JxlPixelFormat` requests it as RGB? Would an RGB profile be supported in this instance? I'd imagine this should work for XYB since grayscale is just XYB with X=B=0, right?

Reply:
Perhaps we go with allowing transformation to any valid RGB and grayscale ICC profile (in order not to make it fail conversion from RGB images while it works for grayscale->grayscale), even though this might mean slighly more work in implementing this.

I'll remove the restriction from the doc string.